ReSharper.FSharp/src/FSharp.Psi.Daemon/src/Analyzers/LambdaAnalyzer.fs

```diff
@@ -84,45 +84,73 @@ type LambdaAnalyzer() =
 
         compareArgsRec expr 0 null
 
-    let isImplicitlyConvertedToDelegate (lambda: ILambdaExpr) =
+    let getWarningIfApplicable (ctor: ILambdaExpr * 'a -> #IHighlighting) (lambda: ILambdaExpr, exprToReplace: 'a as arg) isFSharp6Supported =
         let lambda = lambda.IgnoreParentParens()
         let binaryExpr = BinaryAppExprNavigator.GetByRightArgument(lambda)
         let argExpr = if isNull binaryExpr then lambda else binaryExpr :> _
         let appTuple = TupleExprNavigator.GetByExpression(argExpr)
         let app = getArgsOwner argExpr
 
         let reference = getReference app
-        app :? IPrefixAppExpr && isNotNull reference &&
+        if not (app :? IPrefixAppExpr) || isNull reference then ctor arg else
 
         match reference.GetFcsSymbol() with
-        | :? FSharpMemberOrFunctionOrValue as m ->
-            m.IsMember &&
+        | :? FSharpMemberOrFunctionOrValue as m when m.IsMember ->
             let lambdaPos = if isNotNull appTuple then appTuple.Expressions.IndexOf(argExpr) else 0
 
             let parameterGroups = m.CurriedParameterGroups
-            if parameterGroups.Count = 0 then false else
+            if parameterGroups.Count = 0 then ctor arg else
 
             let args = parameterGroups[0]
-            if args.Count <= lambdaPos then false else
+            if args.Count <= lambdaPos then ctor arg else
 
             let argDecl = args[lambdaPos]
             let argDeclType = argDecl.Type
-
             let argIsDelegate =
                 argDeclType.HasTypeDefinition && (getAbbreviatedEntity argDeclType.TypeDefinition).IsDelegate
-
-            if argIsDelegate then
-                let mApparentEntity = m.ApparentEnclosingEntity
-                let mName = m.DisplayName
-                let mHasOverload =
-                    mApparentEntity.MembersFunctionsAndValues
-                    |> Seq.exists (fun x ->
-                        not (x.Equals m) &&
-                        x.DisplayName = mName &&
-                        x.CurriedParameterGroups[0].Count >= args.Count)
-                not mHasOverload
-            else false
-        | _ -> false
+                
+            // If the lambda is passed instead of a delegate,
+            // then in F# < 6.0 there is almost never an implicit cast for the lambda simplification 
+            if argIsDelegate && not isFSharp6Supported then null else
+
+            match exprToReplace.As<IFSharpExpression>().IgnoreInnerParens() with
+            | :? IReferenceExpr as ref ->
```

Comment:
Maybe use something like this to reduce nesting? Would be cool if there was an easy way to split the logic for existing expressions and new expression names, so we don't do these kind of type checks.
```fsharp
let refExpr = exprToReplace.As<IFSharpExpression>().IgnoreInnerParens().As<IReferenceExpr>()
if isNul refExpr then ctor arg else

...
```

If `GetMethods` if refactored out successfully, then this match expression could become small enough to split and to repeat the util function usage.